src/pybind/mgr/dashboard/frontend/src/app/shared/models/osd-deployment-options.ts

```diff
@@ -0,0 +1,18 @@
+export type DeploymentOptionNames = 'cost-capacity' | 'throughput' | 'iops';
+
+export interface DeploymentOption {
+  name: DeploymentOptionNames;
+  capacity: number;
+  available: boolean;
+  hdd_used: number;
+  used: number;
+  nvme_used: number;
+  ssd_used: number;
+}
+
+export interface DeploymentOptions {
+  options: {
+    [key in DeploymentOptionNames]: DeploymentOption;
+  };
```

Comment:
This looks like an enum to me. And mixing an enum with the selected option is a bit mixing too many concerns all together.

Reply:
Done!